### ISSUE TITLE
EvalSpec's param throttle_secs in tf.estimator.train_and_evaluate

### ISSUE THREAD

**Issue by Bakaizhamgyrchiev** [tensorflow/tensorflow#21896]
### System information
- **Have I written custom code (as opposed to using a stock example script provided in TensorFlow)**: no
- **OS Platform and Distribution (e.g., Linux Ubuntu 16.04)**:  Linux Ubuntu 16.04)
- **Mobile device (e.g. iPhone 8, Pixel 2, Samsung Galaxy) if the issue happens on mobile device**:
- **TensorFlow installed from (source or binary)**:
- **TensorFlow version (use command below)**: 1.10.0
- **Python version**: 3.6
- **Bazel version (if compiling from source)**:
- **GCC/Compiler version (if compiling from source)**:
- **CUDA/cuDNN version**:
- **GPU model and memory**:
- **Exact command to reproduce**:

I use [tf.Estimator](https://www.tensorflow.org/api_docs/python/tf/estimator/Estimator) for training my models.
Here is part of my source code: 
```
train_spec = tf.estimator.TrainSpec(input_fn=lambda: input_fn(.....))

eval_spec = tf.estimator.EvalSpec(input_fn=lambda: input_fn(...), throttle_secs=NUM_OF_SECONDS)

tf.estimator.train_and_evaluate(estimator=estimator, train_spec=train_spec, eval_spec=eval_spec)
```

We can see this part of logs after start of training in Tensorflow 1.9.0:
```
INFO:tensorflow:Running training and evaluation locally (non-distributed).
INFO:tensorflow:Start train and evaluate loop. The evaluate will happen after MANY secs (eval_spec.throttle_secs) or training is finished
```

in Tensorflow 1.10.0
```
INFO:tensorflow:Running training and evaluation locally (non-distributed).
INFO:tensorflow:Start train and evaluate loop. The evaluate will happen after every checkpoint. Checkpoint frequency is determined based on RunConfig arguments: save_checkpoints_steps None or save_checkpoints_secs 600.
```
My question is how does  param throtlle_secs work now? Because, there is reason to start evaluation process after many steps (for example 1M steps). In previous versions set many seconds to throttle_secs helps me. But in 1.10.0 set many seconds to save_checkpoints_secs is risky.

and how does new release [Improved local run behavior in tf.estimator.train_and_evaluate which does not reload checkpoints for evaluation.](https://github.com/tensorflow/tensorflow/blob/master/RELEASE.md) influence on these changes in 1.10.0 version

**Comment by Bakaizhamgyrchiev** [comment]
@cy89 
No, issue #17650 doesn't help. I'm now interested how does throttle secs of EvalSpec works in tf 1.10? 

please, look at tensorflow logs when estimator start to train_and_evaluate in tf 1.9 and 1.10  (look my issue head). It was comfortable to control eval frequency with throttle_secs. Control with save_checkpoints_secs is very risky, when I want start eval process every epoch

**Comment by cy89** [comment]
@martinwicke @ispirmustafa could you comment, please?

**Comment by ispirmustafa** [comment]
Hi @Bakaizhamgyrchiev 
That's a missing part in documentation. We do use `throttle_secs` in local-run too. I'm curious have you tried it?

**Comment by tensorflowbutler** [comment]
It has been 14 days with no activity and the `awaiting response` label was assigned. Is this still an issue?

**Comment by Bakaizhamgyrchiev** [comment]
Hi @ispirmustafa Yes, I tried it, of course. But it doesn't work. I set to throttle_secs = 1000000 secs. But evaluation started after 600 secs (save_checkpoints_secs). Train process works about 6 hours.

**Comment by ispirmustafa** [comment]
Hi @Bakaizhamgyrchiev 
First evaluation should start with first checkpoint. But the second one should be at least `throttle_secs` after that. Could you please verify that?

**Comment by tensorflowbutler** [comment]
Nagging Assignee @cy89: It has been 14 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @cy89: It has been 29 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by Bakaizhamgyrchiev** [comment]
Thanks very much for answer. I have also noticed that if evaluation process is longer than throttle_secs then evaluations will restart without training steps. I thought that throttle_secs was counted down after evaluation process was finished (throttle_secs really is counted down after checkpoint saving). Would be great if it was clear in documentation...

**Comment by rex-yue-wu** [comment]
Same issue. As today, using tensorflow v1.15 and the latest object detection API, I still encounter the same issues:

1. when `throttle_secs` < the time required for validation, no training will happen and the `object_detection/model_main.py` script simply does validation all the time
2. when `throttle_secs` is explicitly set to be greater than the required validation time, training happens in between of two validations, but the actual time gap between two validations does not greater than the time since the previous checkpoint is saved
